crates/fluvio-types/src/event.rs

```diff
@@ -9,47 +9,69 @@ const DEFAULT_EVENT_ORDERING: Ordering = Ordering::SeqCst;
 #[deprecated(since = "0.2.5", note = "use StickyEvent instead")]
 pub use StickyEvent as SimpleEvent;
 
+#[derive(Clone)]
 pub struct StickyEvent {
+    inner: Arc<StickyInner>,
+}
+
+struct StickyInner {
     flag: AtomicBool,
     event: Event,
 }
 
+impl StickyInner {
+    fn is_set(&self) -> bool {
+        self.flag.load(DEFAULT_EVENT_ORDERING)
+    }
+}
+
 impl StickyEvent {
+    #[deprecated(since = "0.2.6", note = "use 'StickyEvent::new()' instead")]
     pub fn shared() -> Arc<Self> {
-        Arc::new(Self {
-            flag: AtomicBool::new(false),
-            event: Event::new(),
-        })
+        Arc::new(Self::new())
+    }
+
+    pub fn new() -> Self {
+        Self {
+            inner: Arc::new(StickyInner {
+                flag: AtomicBool::new(false),
+                event: Event::new(),
+            }),
+        }
     }
 
     // is flag set
     pub fn is_set(&self) -> bool {
-        self.flag.load(DEFAULT_EVENT_ORDERING)
+        self.inner.is_set()
     }
 
-    pub async fn listen(&self) {
-        if self.is_set() {
-            trace!("before, flag is set");
-            return;
-        }
+    pub fn listen(&self) -> impl std::future::Future<Output = ()> {
```

Comment:
use `Deref` in case, this can be move deleted to inner

Reply:
This actually needs to be implemented on the outer type. If you look in the body here, we create an `async move` block and pass an owned `Arc<StickyInner>` into it, that way the Future has the `'static` bound. We wouldn't be able to do this if we put this impl on `StickyInner`.

Reply:
What is we want to achieve here?  not clear how this improve from just simple aliasing.